turbinia/workers/file_system_timeline.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+# Copyright 2022 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Task to run dfimagetools FileEntryLister on disk partitions."""
+
+from __future__ import unicode_literals
+
+import os
+
+from turbinia import TurbiniaException
+from turbinia.workers import TurbiniaTask
+from turbinia.evidence import EvidenceState as state
+from turbinia.evidence import BodyFile
+
+if TurbiniaTask.check_worker_role():
+  try:
+    from dfvfs.helpers import volume_scanner
+    from dfimagetools import file_entry_lister
+  except ImportError as exception:
+    message = 'Could not import libraries: {0!s}'.format(exception)
+    raise TurbiniaException(message)
+
+
+class FileSystemTimelineTask(TurbiniaTask):
+
+  REQUIRED_STATES = [state.ATTACHED]
+
+  def run(self, evidence, result):
+    """Task to execute (dfimagetools) FileEntryLister.
+
+    Args:
+        evidence (Evidence object):  The evidence we will process.
+        result (TurbiniaTaskResult): The object to place task results into.
+
+    Returns:
+        TurbiniaTaskResult object.
+    """
+    bodyfile_output = os.path.join(self.output_dir, 'file_system.bodyfile')
+    output_evidence = BodyFile(source_path=bodyfile_output)
+    number_of_entries = 0
+
+    # Set things up for the FileEntryLister client. We will scan all
+    # partitions in the volume.
+    volume_scanner_options = volume_scanner.VolumeScannerOptions()
+    volume_scanner_options.partitions = ['all']
+
+    # Create the FileEntryLister client and generate the path specs
+    # for all available partitions.
+    entry_lister = file_entry_lister.FileEntryLister()
```

Comment:
This code all LGTM, but I don't really know the dfVFS/dfimagetools code very well.  Maybe @joachimmetz or @dfjxs could also take a quick look at just this part?

Reply:
I chatted with jleaniz offline and most of this is directly adapted from the upstream script, so it probably doesn't need additional review.